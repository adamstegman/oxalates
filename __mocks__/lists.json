[
  {
    "id": "all",
    "name": "All",
    "bottomThreshold": null,
    "topThreshold": null
  },
  {
    "id": 1,
    "name": "Very High",
    "bottomThreshold": 50,
    "topThreshold": null
  },
  {
    "id": 2,
    "name": "High",
    "bottomThreshold": 15,
    "topThreshold": 50
  },
  {
    "id": 3,
    "name": "Moderate",
    "bottomThreshold": 5,
    "topThreshold": 15
  },
  {
    "id": 4,
    "name": "Low",
    "bottomThreshold": null,
    "topThreshold": 5
  }
]
