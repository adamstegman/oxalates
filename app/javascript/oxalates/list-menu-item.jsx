import React from 'react';
import PropTypes from 'prop-types';

import { listPropType } from './list-prop-type';

import './list-menu-item.scss';

const ACTIVE_CLASS_NAME = 'active';
const EMPTY_CLASS_NAME = '';

export class ListMenuItem extends React.Component {
  activeClassName(list) {
    if (this.props.active) {
      return ACTIVE_CLASS_NAME;
    }
    return EMPTY_CLASS_NAME;
  }

  render() {
    return (
      <li className={`list-menu-item ${this.activeClassName(this.props.list)}`}>
        {/* <%= link_to list do %> */}
          <div className="list-menu-item-content">
            <div className="list-menu-item-icon">
              {/* <%= render "list_menu_item_icon", list: list %> */}
            </div>
            <p className="list-menu-item-text">{this.props.list.name}</p>
          </div>
        {/* <% end %> */}
      </li>
    );
  }
};

ListMenuItem.propTypes = {
  list: listPropType.isRequired,
  active: PropTypes.bool,
};
