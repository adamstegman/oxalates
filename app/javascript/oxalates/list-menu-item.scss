$menu_text_color: #aaa;
$menu_active_text_color: darken($menu_text_color, 20%);
$menu_hover_text_color: #000;

$menu_font_size: 10px;

.list-menu-item {
  height: 100%;
  width: 100%;
  a {
    display: block;
    height: 100%;
  }

  a {
    color: $menu_text_color;
    text-decoration: none;

    &:hover, &:focus {
      color: $menu_hover_text_color;
    }
  }
  color: $menu_text_color;
  font-size: $menu_font_size;
  text-align: center;

  &.active {
    color: $menu_active_text_color;
    a {
      color: $menu_active_text_color;

      &:hover, &:focus {
        color: $menu_hover_text_color;
      }
    }
  }
}

.list-menu-item-content {
  height: 100%;

  display: -webkit-flex;
  -webkit-align-items: center;
  -webkit-flex-direction: column;
  -webkit-justify-content: space-around;
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: space-around;
}

.list-menu-item-icon {
  min-height: 24px;
  width: 21px;
}
