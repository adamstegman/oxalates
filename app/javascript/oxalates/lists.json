[
  {
    "id": "all",
    "name": "All",
    "bottom_threshold": null,
    "top_threshold": null
  },
  {
    "id": 1,
    "name": "Very High",
    "bottom_threshold": 50,
    "top_threshold": null
  },
  {
    "id": 2,
    "name": "High",
    "bottom_threshold": 15,
    "top_threshold": 50
  },
  {
    "id": 3,
    "name": "Moderate",
    "bottom_threshold": 5,
    "top_threshold": 15
  },
  {
    "id": 4,
    "name": "Low",
    "bottom_threshold": null,
    "top_threshold": 5
  }
]
