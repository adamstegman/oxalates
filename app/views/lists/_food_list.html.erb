<ul class="food-list">
<% foods.each do |food| %>
  <li class="food-list-item <%= list_class_name(food.list) %>"><%= food.name %></li>
<% end %>
</ul>
