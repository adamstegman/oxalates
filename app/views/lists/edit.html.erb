<%= form_tag list_path(@list.id), method: :put do %>
  <ul class="food-list">
  <% @list.foods.each do |food| %>
    <li>
      <%= link_to food.name, edit_food_path(food), class: "edit-food-list-item #{list_class_name(food.list)}" %>
      <%= hidden_field_tag 'list[foods][][id]', food.id %>
      <%= label_tag 'list[foods][][name]', "Name", class: 'food-list-item-name' %>
      <%= text_field_tag 'list[foods][][name]', food.name, class: 'food-list-item-name' %>
    </li>
  <% end %>
  </ul>
<% end %>
