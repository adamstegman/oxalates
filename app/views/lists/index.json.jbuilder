json.lists(@lists) do |list|
  json.(list, :id, :name, :bottom_threshold, :top_threshold)
end
