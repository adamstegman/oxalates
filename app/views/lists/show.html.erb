<%= render "food_list", foods: @list.foods %>
