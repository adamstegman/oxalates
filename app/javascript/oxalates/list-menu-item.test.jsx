import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import lists from './lists.json';
import { ListMenuItem } from './list-menu-item';

Enzyme.configure({ adapter: new Adapter() });

const noOp = () => {};
const list = lists[0];

test('ListMenuItem renders an inactive list', () => {
  const component = renderer.create(
    <ListMenuItem list={list} active={false} onSelect={noOp} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('ListMenuItem renders an active list', () => {
  const component = renderer.create(
    <ListMenuItem list={list} active={true} onSelect={noOp} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('ListMenuItem calls the callback on selection', () => {
  let selected = false;
  const onSelect = () => {
    selected = true;
  };
  const component = shallow(
    <ListMenuItem list={list} active={true} onSelect={onSelect} />,
  );
  component.find('button').simulate('click');
  expect(selected).toBe(true);
});
