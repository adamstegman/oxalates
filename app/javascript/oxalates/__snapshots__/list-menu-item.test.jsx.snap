// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListMenuItem renders an active list 1`] = `
<li
  className="list-menu-item active"
>
  <button
    className="list-menu-item-content no-outline"
    onClick={[Function]}
    type="button"
  >
    <div
      className="list-menu-item-icon"
    >
      <svg
        height="25"
        width="21"
      >
        <rect
          fill="none"
          height="25"
          stroke="black"
          strokeWidth="3"
          width="21"
          x="0"
          y="0"
        />
        <line
          stroke="black"
          strokeWidth="2"
          x1="3"
          x2="18"
          y1="5"
          y2="5"
        />
        <line
          stroke="#e00"
          strokeWidth="2"
          x1="3"
          x2="18"
          y1="10"
          y2="10"
        />
        <line
          stroke="#fa0"
          strokeWidth="2"
          x1="3"
          x2="18"
          y1="15"
          y2="15"
        />
        <line
          stroke="green"
          strokeWidth="2"
          x1="3"
          x2="18"
          y1="20"
          y2="20"
        />
      </svg>
    </div>
    <p
      className="list-menu-item-text"
    >
      All
    </p>
  </button>
</li>
`;

exports[`ListMenuItem renders an inactive list 1`] = `
<li
  className="list-menu-item "
>
  <button
    className="list-menu-item-content no-outline"
    onClick={[Function]}
    type="button"
  >
    <div
      className="list-menu-item-icon"
    >
      <svg
        height="25"
        width="21"
      >
        <rect
          fill="none"
          height="25"
          stroke="black"
          strokeWidth="3"
          width="21"
          x="0"
          y="0"
        />
        <line
          stroke="black"
          strokeWidth="2"
          x1="3"
          x2="18"
          y1="5"
          y2="5"
        />
        <line
          stroke="#e00"
          strokeWidth="2"
          x1="3"
          x2="18"
          y1="10"
          y2="10"
        />
        <line
          stroke="#fa0"
          strokeWidth="2"
          x1="3"
          x2="18"
          y1="15"
          y2="15"
        />
        <line
          stroke="green"
          strokeWidth="2"
          x1="3"
          x2="18"
          y1="20"
          y2="20"
        />
      </svg>
    </div>
    <p
      className="list-menu-item-text"
    >
      All
    </p>
  </button>
</li>
`;
